### PR TITLE
[SDESK-7381] Support source and parent async resources

### PR DESCRIPTION
**Pull request by MarkLark86**
### Purpose
Provides the following features to async resources:
1. allow multiple [ResourceConfig](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/model.py#L175) to point to the same MongoDB collection and Elasticsearch index
2. ability for a [ResourceRestEndpoint](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/resource_rest_endpoints.py#L55) URL to contain a parent resource (parent/child relationship).

### What has changed
1. New `datasource_name` attribute added to the ResourceConfig dataclass
2. New `parent_links` attribute added to the `rest_endpoints` attribute of the ResourceConfig dataclass

### Documentation
Documentation has been added to the [readthedocs](https://superdesk--2694.org.readthedocs.build/en/2694/core/web.html#resource-rest-endpoint-with-parent) with an example usage and brief explanation

Resolves: [SDESK-7381](https://sofab.atlassian.net/browse/SDESK-7381)

[SDESK-7381]: https://sofab.atlassian.net/browse/SDESK-7381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by MarkLark86** [comment]
@eos87 For example usage, please see the [resource_parent_links](https://github.com/superdesk/superdesk-core/blob/45645ccd08aa2784297353f8fc5e04f06b3b2591/tests/core/resource_parent_links_test.py) test file

File: pyproject.toml [hunk 1]
```diff
@@ -21,3 +21,4 @@ exclude = '''
 testpaths = ["tests", "superdesk", "apps", "content_api"]
 python_files = "*_test.py *_tests.py test_*.py tests_*.py tests.py test.py"
 asyncio_mode = "auto"
+asyncio_default_fixture_loop_scope = "function"
```
Comment (eos87) [inline pyproject.toml:24]: Interesting discussion about this setting https://github.com/pytest-dev/pytest-asyncio/issues/924

**Review by eos87 — APPROVED** [review]
Looks good to me.

File: pyproject.toml (hunk 1 above)
Comment (MarkLark86) [reply pyproject.toml:24]: Yeah I read that, seems silly to me!

**Pull request merged**